### ISSUE TITLE
On Play action we need to do "Save and Refresh"

### ISSUE THREAD

**Issue by van800** [JetBrains/resharper-unity#1205]
Currently we have a Play/Edit action in Rider.
But when you do some changes to code, you should first Save. This Save causes Refresh in Unity. Protocol connection to Unity may be lost for a while. And only after that you can finally call Play.

We need to do it all automatically on Play.

The same thing is required for Attach Debug and Play run configuration.
Debug is calling Save, which causes Refresh, which brakes protocol connection for a short time and Play signal is lost during this short time.

Request origin - private slack channel https://jetbrains.slack.com/archives/CC0R4EJ7P/p1558934359028000

**Comment by van800** [comment]
https://youtrack.jetbrains.com/issue/RIDER-32754

**Comment by ohin** [comment]
May I ask what happened with this issue? For me really is the "Play" button in Rider useless, as I have to go to Unity to check the changes were compiled before starting Play mode anyway...

**Comment by van800** [comment]
It seems that this issue concerns only very small amount of people. So it is not getting prioritized.

@ohin you may always hit Ctrl+S in Rider, it would cause Refresh in Unity, and then Play.

**Comment by ohin** [comment]
Thanks for you reply! I really wonder what is the workflow of others as it seems like, indeed, not many people are concerned :-)

I guess you understand the point, that I have to wait twice - press Ctrl+S (or switch to Unity to trigger recompile) -> wait until it compiles, then press Play -> wait until the game launches. I feel like weirdo now, being alone seeing this as an issue. 

Thanks for listening though :-D

**Comment by CSchnackenberg** [comment]
Agreed. Would be nice if there could be an option: "Save All and Compile before Play in Unity" or similar.
